c10/xpu/impl/XPUGuardImpl.h

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <c10/core/DeviceGuard.h>
+#include <c10/core/impl/DeviceGuardImplInterface.h>
+#include <c10/xpu/XPUCachingAllocator.h>
+#include <c10/xpu/XPUFunctions.h>
+#include <c10/xpu/XPUStream.h>
+
+#include <vector>
+
+namespace c10::xpu::impl {
+
+struct XPUGuardImpl final : public c10::impl::DeviceGuardImplInterface {
+  static constexpr DeviceType static_type = DeviceType::XPU;
```

Comment:
Is this ever used?

Reply:
See https://github.com/pytorch/pytorch/pull/118523/files#diff-5d9aeb30b6a63c8d042d98b533be349b8c451a9b494735beec257a8f6b07aa6bR5
We register `XPUGuardImpl` to `c10::DeviceGuard` so that the user can create a device guard with `XPU` device type.

Reply:
Sure but I don't see any use of static_type in that macro definition?

Reply:
It can be used in line 123 in InlineDeviceGuard. see https://github.com/pytorch/pytorch/blob/91994684016ced9c2124bb16a39fd15e1fc2debb/c10/core/impl/InlineDeviceGuard.h#L117-L125